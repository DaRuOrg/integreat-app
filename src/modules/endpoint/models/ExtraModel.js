class ExtraModel {
  constructor ({name, type, url}) {
    this._name = name
    this._type = type
    this._url = url
  }

  get name () {
    return this._name
  }

  get type () {
    return this._type
  }

  get url () {
    return this._url
  }
}

export default ExtraModel
