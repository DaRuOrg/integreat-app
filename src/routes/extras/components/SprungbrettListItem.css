@import url('../../../modules/common/constants/colors.css');

.job {
  display: flex;
  border-bottom: 2px solid var(--theme-color);
}

.description {
  margin-left: 15px;
  padding: 15px 0;
}

.title {
  font-weight: 700;
}

.horizontal-line {
  border-bottom: 2px solid var(--theme-color);
}
