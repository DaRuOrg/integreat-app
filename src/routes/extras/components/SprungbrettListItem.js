import React from 'react'
import PropTypes from 'prop-types'

import SprungbrettJobModel from 'modules/endpoint/models/SprungbrettJobModel'

class SprungbrettListItem extends React.Component {
  static propTypes = {
    job: PropTypes.instanceOf(SprungbrettJobModel).isRequired
  }

  render () {
    const job = this.props.job

    return <a href={job.url}>
      <div>{job.title}</div>
    </a>
  }
}

export default SprungbrettListItem
