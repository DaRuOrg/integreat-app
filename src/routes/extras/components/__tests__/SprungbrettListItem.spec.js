import React from 'react'
import { shallow } from 'enzyme'
import ExtraModel from '../../../../modules/endpoint/models/ExtraModel'
import SprungbrettListItem from '../SprungbrettListItem'

describe('SprungbrettListItem', () => {
  const extra = new ExtraModel({type: 'ige-ipb', path: 'praktikumsboerse', name: 'IHK PB', thumbnail: 'xy'})

  it('should match snapshot', () => {
    const wrapper = shallow(<SprungbrettListItem job={extra} />)
    expect(wrapper).toMatchSnapshot()
  })
})
