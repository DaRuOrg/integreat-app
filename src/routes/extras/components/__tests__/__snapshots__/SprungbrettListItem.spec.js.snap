// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SprungbrettListItem should match snapshot 1`] = `
<a
  className="job"
>
  <div
    className="title"
  />
  <div
    className="description"
  />
</a>
`;
